components/TextFields/src/MDCTextInputControllerOutlinedTextArea.m

```diff
@@ -86,13 +86,35 @@ - (void)textInputDidLayoutSubviews {
   [self updateBorder];
 }
 
+// clang-format off
+/**
+ textInsets: is the source of truth for vertical layout. It's used to figure out the proper
+ height and also where to place the placeholder / text field.
+
+ NOTE: It's applied before the textRect is flipped for RTL. So all calculations are done here à la
+ LTR.
+
+ The vertical layout is, at most complex, this form:
+
+ MDCTextInputTextFieldOutlinedTextAreaHalfPadding +                   // Small padding
+ MDCTextInputTextFieldOutlinedTextAreaPaddingAdjustment               // Additional point (iOS specific)
+ placeholderEstimatedHeight                                           // Height of placeholder
+ MDCTextInputTextFieldOutlinedTextAreaHalfPadding +                   // Small padding
+ MDCTextInputTextFieldOutlinedTextAreaPaddingAdjustment               // Additional point (iOS specific)
+ MDCCeil(MAX(self.textInput.font.lineHeight,                          // Text field or placeholder
+             self.textInput.placeholderLabel.font.lineHeight))
+ MDCTextInputTextFieldOutlinedTextAreaHalfPadding                     // Padding to bottom of border rect
+ underlineLabelsOffset                                                // From super class.
```

Comment:
Nit: s/underlineLabelsOffset/underlineOffset/

based on the code

Reply:
Updated with more clarity.